src/components/common/copyable/copy-overlay.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { RefObject, useCallback, useEffect, useState } from 'react'
+import { Overlay, Tooltip } from 'react-bootstrap'
+import { Trans, useTranslation } from 'react-i18next'
+import { ShowIf } from '../show-if/show-if'
+
+export interface CopyOverlayProps {
+  content: string
+  clickComponent: RefObject<HTMLElement>
+}
+
+export const CopyOverlay: React.FC<CopyOverlayProps> = ({ content, clickComponent }) => {
+  useTranslation()
+  const [showCopiedTooltip, setShowCopiedTooltip] = useState(false)
+  const [error, setError] = useState(false)
+  const [clickComponentSaved, setClickComponentSaved] = useState<HTMLElement>()
+
+  const copyToClipboard = useCallback((content: string) => {
+    navigator.clipboard.writeText(content).then(() => {
+      setError(false)
+    }).catch(() => {
+      setError(true)
+      console.error("couldn't copy")
+    }).finally(() => {
+      setShowCopiedTooltip(true)
+      setTimeout(() => { setShowCopiedTooltip(false) }, 2000)
+    })
+  }, [])
+
+  useEffect(() => {
+    if (clickComponent && clickComponent.current) {
+      clickComponent.current.addEventListener('click', () => copyToClipboard(content))
+      setClickComponentSaved(clickComponent.current)
+    }
+    return () => {
+      if (clickComponentSaved) {
+        clickComponentSaved.removeEventListener('click', () => copyToClipboard(content))
+      }
+    }
+  }, [clickComponent, copyToClipboard, content, clickComponentSaved])
+
+  return (
+    <Overlay target={clickComponent} show={showCopiedTooltip} placement="top">
+      {(props) => (
+        <Tooltip id={'copied_' + content} {...props}>
```

Comment:
Is it a good idea to use the content as part of the id without any kind of serialization/hashing? I can imagine that this results in errors if non-ascii or other special characters are part of the content.

Reply:
how about `copied_${uuid}`?

Reply:
done